src/test/javascript/spec/component/orion/orion-exercise-details-student-actions.component.spec.ts

```diff
@@ -48,86 +40,88 @@ describe('OrionExerciseDetailsStudentActionsComponent', () => {
+                ideStateStub = jest.spyOn(orionConnector, 'state');
             });
     });
+
     afterEach(() => {
-        sinon.restore();
+        jest.restoreAllMocks();
     });
 
     it('should not reflect that the represented exercise is opened if another exercise has been opened', fakeAsync(() => {
         const stateObservable = new BehaviorSubject(ideState);
         comp.exercise = exercise;
-        ideStateStub.returns(stateObservable);
+        ideStateStub.mockReturnValue(stateObservable);
 
         comp.ngOnInit();
         fixture.detectChanges();
         tick();
 
-        expect(comp.orionState.opened).to.not.equal(exercise.id);
+        expect(comp.orionState.opened).not.toBe(exercise.id);
 
         fixture.destroy();
         flush();
     }));
     it('should reflect that the represented exercise is opened if the same exercise is open in the IDE', fakeAsync(() => {
         const stateObservable = new BehaviorSubject({ opened: exercise.id });
         comp.exercise = exercise;
-        ideStateStub.returns(stateObservable);
+        ideStateStub.mockReturnValue(stateObservable);
 
         comp.ngOnInit();
         fixture.detectChanges();
         tick();
 
-        expect(comp.orionState.opened).to.equal(exercise.id);
+        expect(comp.orionState.opened).toBe(exercise.id);
 
         fixture.destroy();
         flush();
     }));
     it('should clone the correct repository in the IDE', () => {
         const participation = { id: 123, repositoryUrl: 'testUrl' } as ProgrammingExerciseStudentParticipation;
-        const progExercise = { id: 42, title: 'Test Title' } as Exercise;
-        progExercise.studentParticipations = [participation];
-        comp.exercise = progExercise;
+        const programmingExercise = { id: 42, title: 'Test Title' } as Exercise;
+        programmingExercise.studentParticipations = [participation];
+        comp.exercise = programmingExercise;
         comp.courseId = 456;
 
         comp.importIntoIDE();
-        expect(cloneSpy).to.have.been.calledOnceWithExactly('testUrl', progExercise);
+        expect(cloneSpy).toHaveBeenCalledTimes(1);
+        expect(cloneSpy).toHaveBeenCalledWith('testUrl', programmingExercise);
     });
     it('should submit the changes and then forward the build results on submit', () => {
         comp.exercise = exercise;
         comp.submitChanges();
 
-        expect(submitSpy).to.have.been.calledOnce;
-        expect(forwardBuildSpy).to.have.been.calledOnce;
-        expect(forwardBuildSpy).to.have.been.calledImmediatelyAfter(submitSpy);
+        expect(submitSpy).toHaveBeenCalledTimes(1);
+        expect(forwardBuildSpy).toHaveBeenCalledTimes(1);
+        // asserts forwardBuild has been called directly after submit
+        expect(forwardBuildSpy.mock.invocationCallOrder[0]).toBe(submitSpy.mock.invocationCallOrder[0] + 1);
     });
     it('isOfflineIdeAllowed should work', () => {
```

Comment:
"should work" is a weak phrasing. Try something like "should be set in the component" or something like that